src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -886,13 +901,13 @@ void AddUsesPermissions (XElement application, TypeDefinitionCache cache)
 				if (!application.Parent.Descendants ("uses-permission").Any (x => (string)x.Attribute (attName) == upa.Name))
 					application.AddBeforeSelf (upa.ToElement (PackageName, cache));
 		}
-		void AddUsesConfigurations (XElement application, IEnumerable<UsesConfigurationAttribute> configs, TypeDefinitionCache cache)
+		void AddUsesConfigurations (XElement application, List<UsesConfigurationAttribute> configs, TypeDefinitionCache cache)
 		{
 			foreach (var uca in configs)
 				application.Add (uca.ToElement (PackageName, cache));
 		}
 
-		void AddUsesLibraries (XElement application, IEnumerable<UsesLibraryAttribute> libraries, TypeDefinitionCache cache)
+		void AddUsesLibraries (XElement application, List<UsesLibraryAttribute> libraries, TypeDefinitionCache cache)
```

Comment:
These might allow the C# compiler to automatically turn the `foreach` into `for`, if the `List<T>` type is known.